processor/processortest/unhealthy_processor.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package processortest // import "go.opentelemetry.io/collector/processor/processortest"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/processor"
+)
+
+// NewUnhealthyProcessorCreateSettings returns a new nop settings for Create*Processor functions.
+func NewUnhealthyProcessorCreateSettings() processor.CreateSettings {
+	return processor.CreateSettings{
+		TelemetrySettings: componenttest.NewNopTelemetrySettings(),
+		BuildInfo:         component.NewDefaultBuildInfo(),
+	}
+}
+
+// NewUnhealthyProcessorFactory returns a component.ProcessorFactory that constructs nop processors.
+func NewUnhealthyProcessorFactory() processor.Factory {
+	return processor.NewFactory(
+		"unhealthy",
+		func() component.Config {
+			return &struct{}{}
+		},
+		processor.WithTraces(createUnhealthyTracesProcessor, component.StabilityLevelStable),
+		processor.WithMetrics(createUnhealthyMetricsProcessor, component.StabilityLevelStable),
+		processor.WithLogs(createUnhealthyLogsProcessor, component.StabilityLevelStable),
+	)
+}
+
+func createUnhealthyTracesProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Traces) (processor.Traces, error) {
+	return &unhealthyProcessor{
+		Consumer:  consumertest.NewNop(),
+		telemetry: set.TelemetrySettings,
+	}, nil
+}
+
+func createUnhealthyMetricsProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Metrics) (processor.Metrics, error) {
+	return &unhealthyProcessor{
+		Consumer:  consumertest.NewNop(),
+		telemetry: set.TelemetrySettings,
+	}, nil
+}
+
+func createUnhealthyLogsProcessor(_ context.Context, set processor.CreateSettings, _ component.Config, _ consumer.Logs) (processor.Logs, error) {
+	return &unhealthyProcessor{
+		Consumer:  consumertest.NewNop(),
+		telemetry: set.TelemetrySettings,
+	}, nil
+}
+
+type unhealthyProcessor struct {
+	component.StartFunc
+	component.ShutdownFunc
+	consumertest.Consumer
+	telemetry component.TelemetrySettings
+}
+
+func (p unhealthyProcessor) Start(_ context.Context, _ component.Host) error {
+	go func() {
+		_ = p.telemetry.ReportComponentStatus(component.NewStatusEvent(component.StatusRecoverableError))
```

Comment:
Would it be useful for the processor to support other events as well?

Reply:
It could be updated to support other events. We use this in a test where we start a collector and observe the results. It is not possible (or at least not easy) to get a handle to a processor to trigger events from a component. We could pass in a static event to the factory, but I don't think we'd take advantage of that functionality. The testcase this is used in verifies that an event make it from a component to a status watcher. It's not necessary to exhaustively test the type of events, we do that at a lower level. In addition, we automatically get `Starting`, `Stopping`, and `Stopped` events from auto status reporting in graph. tl;dr it could be made more generic, but we would not be taking advantage of it currently.

Reply:
That said, if you'd like it made more generic, let me know.

Reply:
It can be improved in the future if the need arises